src/components/NcActionButton/NcActionButton.vue

```diff
@@ -269,6 +381,34 @@ export default {
 			type: Boolean,
 			default: false,
 		},
+
+		/**
+		 * The button's behavior, by default the button acts like a normal button with optional toggle button behavior if `modelValue` is `true` or `false`
+		 * But you can also set to checkbox button behavior with tri-state or radio button like behavior.
+		 */
+		modelBehavior: {
+			type: String,
+			default: 'button',
+			validator: (behavior) => ['checkbox', 'radio', 'button'].includes(behavior),
+		},
```

Comment:
**thought:**

For me, it is not clear what is the difference between `checkbox` and `button` if both supports `modelValue`.

Reply:
If I understand correctly, the only difference is that `checkbox` support explicit `null` `modelValue`.

Reply:
**thought:**

Together with `radioValue`, "string-value" behavior makes me feel that the component is overloaded with long-named props on the minimal usage of a radio:

```html
<NcActionButton :model-value.sync="checked" radio-value="left" model-behavior="radio" />
<NcActionButton :model-value.sync="checked" radio-value="middle" model-behavior="radio" />
<NcActionButton :model-value.sync="checked" radio-value="right" model-behavior="radio" />
```

What about making `radio-value` just `value` and `model-behavior="radio"`/model- behavior="checkbox" simplify to bool props `radio`/`checkbox` flags:

```html
<NcActionButton :model-value.sync="checked" value="left" radio />
<NcActionButton :model-value.sync="checked" value="middle" radio />
<NcActionButton :model-value.sync="checked" value="right" radio />
```

This brings undefined behavior when components have both `radio` and `checkbox`, but I think it doesn't make sense to pass both and won't be confusing. And no flag = just a button as usually.

And I don't see a case where we would need to dynamically change `model-behavior` to another value.

What do you think?

Reply:
I am not sure here, while it is shorter I think it removes some makes it less informative.
It is not bad but I think if we really want to make it short we should do it like with `value` and use a property similar to the native HTML one.

Meaning we could use `type` with `radio`, `checkbox`, `button` just like the `<input>` element.

Reply:
With `type` it mixes with button native `type` and we may come back to ugly `native-type` here again. 

For example, we add dialog support on `NcActions` with a form (we already use it as a dialog form actually) when we want to have a submit button in the `NcActions`.

(The same with `value` actually, but `value` on button is barely used on buttons in JS and I guess many developers even don't remember what button value was for).

Reply:
Makes sense, but then I would stick with `model-behavior` (or just `behavior`?)

Reply:
I just realize that having button native `type` doesn't conflict with a radio/checkbox type. As A button cannot be at the same time e.g. `submit` and `radio`, so when we need `type="submit"` it only requires to add one more type to the list `radio|checkbox|button|submit`

Let's go with `type`?

Reply:
Sounds good! Implemented like that and also forward to `NcButton::nativeType` for inline actions.